src/core/json_object.h

```diff
@@ -0,0 +1,75 @@
+// Copyright 2022, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <memory>
+#include <memory_resource>
+#include <optional>
+#include <string_view>
+
+namespace jsoncons {
+struct sorted_policy;
+template <typename CharT, typename Policy, typename Allocator> class basic_json;
+}  // namespace jsoncons
+
+// the last one in object.h is OBJ_STREAM and it is 6,
+// this will add enough place for Redis types to grow
+constexpr unsigned OBJ_JSON = 15;
+
+namespace dfly {
+
+namespace detail {
+
+// This is temporary, there is an issue right now with jsoncons about using jsonpath
+// with custom allocator. once it would resolved, we would change this to use custom allocator
+// that allocate memory from mimalloc
+using JsonType = jsoncons::basic_json<char, jsoncons::sorted_policy, std::allocator<char>>;
+
+class JsonWrapper {
+ public:
+  using InternalObj = JsonType*;
+
+  JsonWrapper() = default;
+  JsonWrapper(JsonWrapper&&) = default;
+  JsonWrapper& operator=(JsonWrapper&&) = default;
+
+  explicit JsonWrapper(JsonType&& j, std::pmr::memory_resource* mr);
+
+  size_t MallocUsed() const;
+
+  unsigned encoding() const;
+
+  InternalObj inner_obj() const {
+    return json_ptr_;
+  }
+
+  // pre-condition: Set was called
+  // If not it would "panic"
+  const JsonType& AsJson() const;
+
+  bool Equal(const JsonWrapper& ow) const;
+
+  size_t Size() const {
+    return sz_;
+  }
+  void Free(std::pmr::memory_resource* mr);
+
+  void Set(JsonType&& j, std::pmr::memory_resource* mr);  // we start from string
```

Comment:
See my comment above, it would require that someone from the outside would allocate memory from mimalloc and then it would need to free it as well. I tried to make the interface more like the one with RobjWeapper.

Reply:
I think that if we are just passing a pointer, it would be better to remove this type completely and put the code inside CompactObject. Maybe just writing the interface for the standard allocator (which uses the mimallc allocation) just it. This can be a good option as well.

Reply:
I agree.

Reply:
I will push a change where the JsonWrapper is removed (Will move the code to CompactObject. I'm leaving the file json_object because we will need it later in order to add the custom allocator for jsoncons and I don't want to add to the header file of compact object all this code.